File: easy/twosum.py
def twoSum(nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """
    hm = {}
    for i in range(len(nums)):
        dif = target - nums[i]
        if dif in hm:
            return sorted([hm[dif], i])
        else:
            hm[nums[i]] = i





nums = [2, 2, 11, 15, 18]
target = 20
print(twoSum(nums, target))

File: easy/twoSum.cpp
#include <vector>
#include <unordered_map>

std::vector<int> twoSum(std::vector<int>& nums, int target) {
    std::unordered_map<int, int> hm {};

    for (int i = 0; i < nums.size(); i++) {
        auto dif = hm.find(target - nums[i]);

        if (dif != hm.end()) {
            return std::vector<int> {std::min(i, dif->second), std::max(i, dif->second)};
        } else {
            hm.insert({nums[i], i});
        }
    }
    return std::vector<int> {0, 0};
}
